homeassistant/components/fordpass/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for FordPass integration."""
+import logging
+
+from fordpass import Vehicle
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN, VIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(VIN): vol.All(str, vol.Length(min=17, max=17)),
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    vehicle = Vehicle(data[CONF_USERNAME], data[CONF_PASSWORD], data[VIN])
+
+    try:
+        result = await hass.async_add_executor_job(vehicle.auth)
+
+    except Exception as ex:
+        raise InvalidAuth from ex
+
+    if not result:
+        _LOGGER.error("Failed to authenticate with fordpass")
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {"title": f"Vehicle ({data[VIN]})"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for FordPass."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
Move `self.async_create_entry()` out of the try as it doesn't throw an exception.